Make podman-compose refer to podman-compose(1) when using an external provider

marinmo opened the pull request:
This PR is meant to fix https://github.com/containers/podman/issues/22904. It makes the warning message emitted by podman-compose when using an external provider point to podman-compose(1) which is also updated to include instructions on how to disable said message.

#### Does this PR introduce a user-facing change?

```release-note
Changed the warning message emitted by `podman-compose` when using an external compose provider. Instead of referring to "the documentation" it now refers to the man page of `podman-compose(1)` which is also updated to contain the relevant information.
```

marinmo commented:
Seems the build is failing due to tests which really shouldn't be necessary for a man page? Please ltm if any further action is required on my part.

rhatdan commented on cmd/podman/compose.go:
```diff
@@ -202,7 +202,7 @@ func composeProviderExec(args []string, stdout io.Writer, stderr io.Writer, warn
 
 	if warn {
-		fmt.Fprint(os.Stderr, underline(fmt.Sprintf(">>>> Executing external compose provider %q. Please refer to the documentation for details. <<<<\n\n", provider)))
+		fmt.Fprint(os.Stderr, underline(fmt.Sprintf(">>>> Executing external compose provider %q. Please see podman-compose(1) for how to disable this message. <<<<\n\n", provider)))
```
Podman will work with docker-compose also?

marinmo replied:
Beats me, but according to https://linuxconfig.org/how-to-use-docker-compose-with-podman-on-linux there seems to be some kind of use case for it I guess?

Luap99 replied:
`podman-compose(1)` is the proper syntax for our man page. I am not sure if the actual podman-compose program ships a man page but likely not otherwise we would have heard about conflicts by now.

marinmo replied:
I'm no Go expert (by far, at all) but it seems to me fmt.Fprint can only output unformatted text. Thus I did not change this from the original commit. It seems fmt.Fprintf could print formatted text like `podman-compose(1)` or such. Please advise if necessary.

rhatdan commented on docs/source/markdown/podman-compose.1.md.in:
```diff
@@ -13,6 +13,8 @@ The default compose providers are `docker-compose` and `podman-compose`.  If ins
 
 If you want to change the default behavior or have a custom installation path for your provider of choice, please change the `compose_provider` field in `containers.conf(5)`.  You may also set the `PODMAN_COMPOSE_PROVIDER` environment variable.
 
+By default, `podman-compose` will emit a warning when using an external compose provider (such as `docker-compose`). This warning can be disabled by setting `compose_warning_logs` to false in `containers.conf(5)` or setting the `PODMAN_COMPOSE_WARNING_LOGS` environment variable to false. See the man page for `containers.conf(5)` for more information.
```
I believe this should be `podman compose`  podman-compose and docker-compose are both two external commands.

marinmo replied:
Example given in #22904 is /usr/local/bin/docker-compose which is why I used that example. Having podman-compose being referred to as "External compose provider" when invoking podman compose seems ... Confusing, at best, to me - but I'm open to change :) Could it be more vendor neutral? Perhaps something like
"By default, `podman compose` will emit a warning when using an external compose provider. [...]"?

Luap99 commented on docs/source/markdown/podman-compose.1.md.in:
```diff
@@ -13,6 +13,8 @@ The default compose providers are `docker-compose` and `podman-compose`.  If ins
 
 If you want to change the default behavior or have a custom installation path for your provider of choice, please change the `compose_provider` field in `containers.conf(5)`.  You may also set the `PODMAN_COMPOSE_PROVIDER` environment variable.
 
+By default, `podman-compose` will emit a warning when using an external compose provider (such as `docker-compose`). This warning can be disabled by setting `compose_warning_logs` to false in `containers.conf(5)` or setting the `PODMAN_COMPOSE_WARNING_LOGS` environment variable to false. See the man page for `containers.conf(5)` for more information.
```
```suggestion
By default, `podman compose` will emit a warning saying that it executes an external command. This warning can be disabled by setting `compose_warning_logs` to false in `containers.conf(5)` or setting the `PODMAN_COMPOSE_WARNING_LOGS` environment variable to false. See the man page for `containers.conf(5)` for more information.
```

The external provider part is already explained in the two paragraphs above as such I do not think you have to mention it at all. Also it always uses a external provider so saying "when using" is incorrect as this warning always will be there.

marinmo replied:
Updated as suggested

Luap99 commented:
Also please squash the commits and rebase

Luap99 commented:
@marinmo Can you please squash the commits (and fix the missing sign-of line with that)
And make sure you rebase on latest main please.

marinmo commented:
Think that was a rather crude fix on my side, my git skills are lacking unfortunately. Squashed but accidentally left both comments intact. Rebased main (albeit after force pushing - does it matter?).

rhatdan commented:
git commit -a --amend -s
git push --force

Luap99 reviewed (approved):
LGTM

Luap99 commented:
Thank you.

rhatdan commented:
/lgtm
